### PR TITLE
[Images] Build python 3.7 images

### PR DESCRIPTION
**Pull request by AlonMaor14**
Images to upload:
- [x] mlrun/prebaked-cuda:11.0-cudnn8-devel-ubuntu18.04
- [x] mlrun/prebaked-cuda:11.7.0-base-ubuntu20.04 (python 3.9.16)
- [x] quay.io/mlrun/jupyter-scipy-notebook:python-3.9 (jupyter/scipy-notebook:python-3.9)

Another TODOs:
- [x] Revert part of the changes in https://github.com/mlrun/mlrun/pull/2890 (we need to return some of the CI to run on 3.7)

Not specifying python patch version anymore but expecting cache to still be used

### CODE REVIEW

File: dockerfiles/mlrun-api/Dockerfile [hunk 1]
```diff
@@ -11,9 +11,9 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG MLRUN_PYTHON_VERSION=3.9.13
+ARG MLRUN_PYTHON_VERSION=3.9
 
-FROM quay.io/mlrun/python:${MLRUN_PYTHON_VERSION}-slim
+FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}
```
Comment (liranbg) [inline dockerfiles/mlrun-api/Dockerfile:16]:
```suggestion
FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}-slim
```

File: dockerfiles/mlrun/Dockerfile [hunk 2]
```diff
@@ -12,9 +12,9 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG MLRUN_PYTHON_VERSION=3.9.13
+ARG MLRUN_PYTHON_VERSION=3.9
 
-FROM quay.io/mlrun/python:${MLRUN_PYTHON_VERSION}-slim
+FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}
```
Comment (liranbg) [inline dockerfiles/mlrun/Dockerfile:17]:
```suggestion
FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}-slim
```

File: dockerfiles/test-system/Dockerfile [hunk 3]
```diff
@@ -11,9 +11,9 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG MLRUN_PYTHON_VERSION=3.9.13
+ARG MLRUN_PYTHON_VERSION=3.9
 
-FROM quay.io/mlrun/python:${MLRUN_PYTHON_VERSION}-slim
+FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}
```
Comment (liranbg) [inline dockerfiles/test-system/Dockerfile:16]:
```suggestion
FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}-slim
```

File: dockerfiles/test/Dockerfile [hunk 4]
```diff
@@ -11,9 +11,9 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ARG MLRUN_PYTHON_VERSION=3.9.13
+ARG MLRUN_PYTHON_VERSION=3.9
 
-FROM quay.io/mlrun/python:${MLRUN_PYTHON_VERSION}-slim
+FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}
```
Comment (liranbg) [inline dockerfiles/test/Dockerfile:16]:
```suggestion
FROM gcr.io/iguazio/python:${MLRUN_PYTHON_VERSION}-slim
```

**Review by liranbg — CHANGES_REQUESTED** [review]
minor

File: Makefile [hunk 5]
```diff
@@ -72,6 +66,8 @@ MLRUN_DOCKER_CACHE_FROM_FLAG :=
 MLRUN_USE_CACHE := $(if $(MLRUN_NO_CACHE),,true)
 MLRUN_DOCKER_NO_CACHE_FLAG := $(if $(MLRUN_NO_CACHE),--no-cache,)
 MLRUN_PIP_NO_CACHE_FLAG := $(if $(MLRUN_NO_CACHE),--no-cache-dir,)
+MLRUN_ANACONDA_PYTHON_DISTRIBUTION = $(shell echo "$(MLRUN_PYTHON_VERSION)" | awk -F. '{print "-py"$$1$$2}')
```
Comment (Tankilevitch) [inline Makefile:69]: let's add a comment for what is being outputes here

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
Looks good, minor comment for readability, also would be useful to add somewhere comment about the change we done to our base images that we removed the patch version and the version bump will be done in the docker registry.

File: .github/workflows/build-workflow-matrix.json [hunk 6]
```diff
@@ -1,23 +1,57 @@
...
-    "image-name":"api"
+    "image-name":"mlrun",
+    "python-version":"3.7",
+    "include-suffix":"true"
   },
   {
-    "image-name":"base"
+    "image-name":"api",
+    "python-version":"3.9",
+    "include-suffix":""
```
[the remark this answers is not in the document]
Comment (AlonMaor14) [reply .github/workflows/build-workflow-matrix.json:15]: add comment in the makefile because we can't write comments in json files

**Review by Tankilevitch — APPROVED** [review]

**Pull request merged**